packages/flutter/lib/src/painting/borders.dart

```diff
@@ -75,8 +76,33 @@ class BorderSide {
   const BorderSide({
     this.color: const Color(0xFF000000),
     this.width: 1.0,
-    this.style: BorderStyle.solid
-  });
+    this.style: BorderStyle.solid,
+  }) : assert(color != null),
+       assert(width != null),
+       assert(width >= 0.0),
+       assert(style != null);
+
+  /// Creates a [BorderSide] that represents the addition of the two given
+  /// [BorderSide]s.
+  ///
+  /// It is only valid to call this if [canMerge] returns true for the two
+  /// sides.
+  ///
+  /// If both sides are null, then this will return null.
+  static BorderSide merge(BorderSide a, BorderSide b) {
+    assert(canMerge(a, b));
+    if (a == null)
+      return b; // might return null
+    if (b == null)
+      return a;
+    assert(a.color == b.color);
```

Comment:
Aren't these asserts covered by the `canMerge()` assert?  ... not that extra asserts hurt anything...

Reply:
I work on the principle that asserts don't trust the rest of the code, because if we trusted the rest of the code, we wouldn't need any asserts. So this is basically verifying that canMerge isn't busted.